Sources/FoundationEssentials/Formatting/DateComponents+ISO8601FormatStyle.swift

```diff
@@ -0,0 +1,820 @@
+                        
+                        if let maybeColon = it.peek(), maybeColon == UInt8(ascii: ":") {
+                            // Throw it away
+                            it.advance()
+                        }
+
+                        if let secondsTens = it.peek(), isASCIIDigit(secondsTens) {
+                            // We have seconds
+                            let seconds = try it.digits(maxDigits: 2, input: inputString, onFailure: Date.ISO8601FormatStyle(self).format(Date.now))
+                            tzOffset = (hours * 3600) + (minutes * 60) + seconds
+                        } else {
+                            // If the next character is missing, that's allowed - the time can be something like just -0852 and then the string can end
+                            tzOffset = (hours * 3600) + (minutes * 60)
+                        }
+                    }
+                }
+                
+                if tzOffset == 0 {
+                    timeZone = .gmt
+                } else {
+                    guard let parsedTimeZone = TimeZone(secondsFromGMT: positive ? tzOffset : -tzOffset) else {
+                        // Out of range time zone
+                        throw parseError(inputString, exampleFormattedString: Date.ISO8601FormatStyle(self).format(Date.now))
+                    }
+                    
+                    timeZone = parsedTimeZone
+                }
+            }
+        }
+        
+        // Use the internal init which does not attempt to check each value for Int.max
+        let dc = DateComponents(calendar: Calendar(identifier: .iso8601),
+                                timeZone: timeZone,
+                                rawEra: nil,
+                                rawYear: year,
+                                rawMonth: month,
+                                rawDay: day,
+                                rawHour: hour,
+                                rawMinute: minute,
+                                rawSecond: second,
+                                rawNanosecond: nanosecond,
+                                rawWeekday: weekday,
+                                rawWeekdayOrdinal: nil,
+                                rawQuarter: nil,
+                                rawWeekOfMonth: nil,
+                                rawWeekOfYear: weekOfYear,
+                                rawYearForWeekOfYear: yearForWeekOfYear,
+                                rawDayOfYear: dayOfYear)
+
+        // Would be nice to see this functionality on BufferView, but for now we calculate it ourselves.
+        let utf8CharactersRead = it.curPointer - view.startIndex._rawValue
+        return ComponentsParseResult(consumed: utf8CharactersRead, components: dc)
+    }
+}
+
+// MARK: `FormatStyle` protocol membership
+
+@available(FoundationPreview 6.2, *)
+public extension FormatStyle where Self == DateComponents.ISO8601FormatStyle {
+    static var iso8601: Self {
```

Comment:
The proposal has it named `iso8601Components`. I think it's ok to call it `iso8601` here (unlike the one below for `RegexComponent`, but not sure what you ended up deciding on

Reply:
I left this one as iso8601 and made the regex one “components” (and also made that consistent with HTTP)